File: lib/src/main/java/io/github/hidekatsu_izuno/pglite_jdbc/pglite/contrib/ltree.java
package io.github.hidekatsu_izuno.pglite_jdbc.pglite.contrib;

import io.github.hidekatsu_izuno.pglite_jdbc.pglite.interface_.Extension;
import io.github.hidekatsu_izuno.pglite_jdbc.pglite.interface_.ExtensionSetupResult;
import java.net.URL;
import java.util.concurrent.CompletableFuture;

public final class ltree {
    public static final Extension<Object> ltree = new Extension<>();

    static {
        ltree.name = "ltree";
        ltree.setup = (_pg, emscriptenOpts, clientOnly) -> {
            var result = new ExtensionSetupResult<Object>();

            result.bundlePath = resource("ltree.tar.gz");
            return CompletableFuture.completedFuture(result);
        };
    }

    private static URL resource(String name) {
        var url = ltree.class.getClassLoader().getResource(name);
        if (url == null) {
            throw new RuntimeException("Extension bundle not found on classpath: " + name);
        }
        return url;
    }

    private ltree() {
    }
}

File: pglite/src/pglite/contrib/ltree.ts
import type {
	Extension,
	ExtensionSetupResult,
	PGliteInterface,
} from "../interface";

const setup = async (_pg: PGliteInterface, _emscriptenOpts: any) => {
	return {
		bundlePath: new URL("ltree.tar.gz", import.meta.url),
	} satisfies ExtensionSetupResult;
};

export const ltree = {
	name: "ltree",
	setup,
} satisfies Extension;
